smarc_bt/src/mission_plan.py

```diff
@@ -473,6 +539,212 @@ def path_to_list(self, path_msg):
+        :param tangent_tol: Numerical tolerance at which we decide the intersections are close enough to consider it a tangent
+        :return Sequence[Tuple[float, float]]: A list of length 0, 1, or 2, where each element is a point at which the circle intercepts a line segment.
+
+        Note: We follow: http://mathworld.wolfram.com/Circle-LineIntersection.html
+        """
+
+        (p1x, p1y), (p2x, p2y), (cx, cy) = pt1, pt2, circle_center
+        (x1, y1), (x2, y2) = (p1x - cx, p1y - cy), (p2x - cx, p2y - cy)
+        dx, dy = (x2 - x1), (y2 - y1)
+        dr = (dx ** 2.0 + dy ** 2.0)**.5
+        big_d = x1 * y2 - x2 * y1
+        discriminant = circle_radius ** 2.0 * dr ** 2.0 - big_d ** 2.0
+
+        if discriminant < 0:  # No intersection between circle and line
+            return []
+        else:  # There may be 0, 1, or 2 intersections with the segment
+            intersections = [
+                (cx + (big_d * dy + sign * (-1 if dy < 0 else 1) * dx * discriminant**.5) / dr ** 2.0,
+                 cy + (-big_d * dx + sign * abs(dy) * discriminant**.5) / dr ** 2.0)
+                for sign in ((1, -1) if dy < 0 else (-1, 1))]  # This makes sure the order along the segment is correct
+            if not full_line:  # If only considering the segment, filter out intersections that do not fall within the segment
+                fraction_along_segment = [(xi - p1x) / dx if abs(dx) > abs(dy) else (yi - p1y) / dy for xi, yi in intersections]
+                intersections = [pt for pt, frac in zip(intersections, fraction_along_segment) if 0 <= frac <= 1]
+            if len(intersections) == 2 and abs(discriminant) <= tangent_tol:  # If line is tangent to circle, return just one point (as both intersections have same location)
+                return [intersections[0]]
+            else:
+                return intersections
+
+    def gps_fix_cb(self, gps_msg):
+        self.is_fix_ok = True
+
+    def dubins_mission_planner(self, mission, num_points=2, inside_turn=True):
+        ''' 
+        Reads the waypoints from a MissionControl message and generates a sampled dubins 
+        path between them. 
+
+        :param mission: the original MissionControl message
+        :param turn_radius: turning radius of the robot [m]
+        :param num_points: the amount of waypoints to keep on each segment between waypoints
+        :param inside_turn: cut the waypoints corners 
+        :param int_radius: only used if inside_turn is True. Radius of the circle used to compute the intersection waypoints [m]
+        :return MissionControl.msg: new MissionControl message equal to the input one except for the waypoints attribute
+        '''
+
+        # Only continue if we have the gps fix
+        self.is_fix_ok = False
+        rospy.Subscriber("/lolo/core/gps", NavSatFix, self.gps_fix_cb)
```

Comment:
Nice hardcoded sub ;)

Reply:
My bad, I hardcoded it for testing purposes and then I completely forgot to remove it 
If the bt takes care of checking the fix, I can definitely remove it

Reply:
You can simply check if the vehicle object in the BB has a location in it. That is always the "last known pose". In theory, you don't even need to check that, because of that is not there, the BT wont even tick the mission parts, but being defensive never hurt anyone :)